modules/services/deeplink/src/main/kotlin/au/com/shiftyjelly/pocketcasts/deeplink/DeepLink.kt

```diff
@@ -0,0 +1,22 @@
+package au.com.shiftyjelly.pocketcasts.deeplink
+
+import android.content.Context
+import android.content.Intent
+import au.com.shiftyjelly.pocketcasts.deeplink.DeepLink.Companion.ACTION_OPEN_DOWNLOADS
+
+sealed interface DeepLink {
+    fun toIntent(context: Context): Intent
+
+    companion object {
+        const val ACTION_OPEN_DOWNLOADS = "INTENT_OPEN_APP_DOWNLOADING"
+    }
+}
+
+data object DownloadsDeepLink : DeepLink {
+    override fun toIntent(context: Context) = context.launcherIntent
+        .setAction(ACTION_OPEN_DOWNLOADS)
+}
```

Comment:
Just an idea, but how about including the action name in the object as they are linked?
```suggestion
data object DownloadsDeepLink : DeepLink {
    override fun action() = "INTENT_OPEN_APP_DOWNLOADING"
    override fun toIntent(context: Context) = context.launcherIntent
        .setAction(action())
}
```

Reply:
I think I get the intention but it doesn't work well in my opinion. Some deep links are data classes and not data objects so `action()` cannot be invoked on them without instantiation. In that case adapters that create non-object deep links cannot access actions.

I could move all actions to objects and to companion objects for some better organization if you think it's worth it?

Reply:
No worries, thanks for considering it.